brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/spec/ConfigKeyModifier.java

```diff
@@ -68,7 +97,66 @@ private void configureWithResolvedConfigKey(FlagUtils.FlagConfigKeyAndValueRecor
         }
     }
 
-    private void setUnusedKeysAsAnonymousKeys(EntitySpec spec, Set<String> keyNamesUsed, ConfigBag bag){
+    private Optional<Object> joinOldAndNewSpecConfigValues(Optional<Object> oldValue,
+                                                           Optional<Object> newValue) {
+        Optional<Object> result;
+
+        if ((newValue.isPresent()) && (oldValue.isPresent())) {
+            result = Optional.of(
+                    joinOldAndNewValues(oldValue.get(), newValue.get()));
+        } else if (newValue.isPresent()) {
+            result = newValue;
+        } else if (oldValue.isPresent()) {
+            result = oldValue;
+        } else {
+            result = Optional.absent();
+        }
+        return result;
+    }
+
+    protected Object joinOldAndNewValues(Object oldValue, Object newValue) {
+        if ((oldValue instanceof Map) && (newValue instanceof Map)) {
+            return combineCurrentAndResolvedValueMaps((Map) oldValue, (Map) newValue);
+        } else if ((oldValue instanceof List) && (newValue instanceof List)) {
+            return combineCurrentAndResolvedValueList((List) oldValue, (List) newValue);
+        } else if ((oldValue instanceof String) && (newValue instanceof String)) {
+            return newValue;
+        } else {
+            LOG.warn("Types of oldValue {} and newValue {} are not the same, so {} is not able " +
+                    "to join them. Then, new type will be returned ",
+                    new Object[]{oldValue, newValue, this});
+            return newValue;
+        }
+    }
+    
+    private Map combineCurrentAndResolvedValueMaps(Map currentValue, Map resolvedVaue) {
+        currentValue.putAll(resolvedVaue);
+        return currentValue;
+    }
+
+    private List combineCurrentAndResolvedValueList(List currentValue, List resolvedVaue) {
+        currentValue.addAll(resolvedVaue);
+        return currentValue;
+    }
+
+    private Optional<Object> getConfigKeyValue(EntitySpec spec,
+                                               FlagUtils.FlagConfigKeyAndValueRecord r) {
+        Object configValue = spec.getConfig().get(r.getConfigKey());
```

Comment:
You can just write `return Optional.fromNullable(spec.getConfig().get(r.getConfigKey()))`